scipy/spatial/qhull.pyx

```diff
@@ -2410,3 +2449,269 @@ class Voronoi(_QhullUser):
             self._ridge_dict = dict(zip(map(tuple, self.ridge_points.tolist()),
                                         self.ridge_vertices))
         return self._ridge_dict
+
+#------------------------------------------------------------------------------
+# Halfspace intersection interface, for Python
+#------------------------------------------------------------------------------
+
+class _QhalfUser(object):
```

Comment:
Code duplication. AFAICS, the only change is s/points/halfspaces. If so, the better fix is to rename `add_points -> _add_points` in QhullUser, and add the appropriately named interface method in the user-facing classes.